### ISSUE TITLE
Unmarshaling of Overpass JSON fails on version field

### ISSUE THREAD

**Issue by fawick** [paulmach/osm#44]
The version information is encoded as JSON number instead of a string by Overpass API. This causes unmarshaling it to fail with error `json: cannot unmarshal number into Go struct field .version of type string`

Take, for example, the minimal JSON output of overpass (http://overpass-api.de/api/interpreter?data=[out:json];out;), e.g.

```json
{
  "version": 0.6,
  "generator": "Overpass API 0.7.59 e21c39fe",
  "osm3s": {
    "timestamp_osm_base": "2022-11-17T17:14:39Z",
    "copyright": "The data included in this document is from www.openstreetmap.org. The data is made available under ODbL."
  },
  "elements": []
}
```

Minimum viable example failure: https://go.dev/play/p/n7kBY3CCbHC

```go
package main

import (
	"encoding/json"
	"fmt"

	"github.com/paulmach/osm"
)

func main() {
	var overpass osm.OSM
	buf := []byte(`{
  "version": 0.6,
  "generator": "Overpass API 0.7.59 e21c39fe",
  "osm3s": {
    "timestamp_osm_base": "2022-11-17T17:14:39Z",
    "copyright": "The data included in this document is from www.openstreetmap.org. The data is made available under ODbL."
  },
  "elements": []
}`)
	err := json.Unmarshal(buf, &overpass)
	if err != nil {

		fmt.Println(err)
	}
}
```

**Comment by paulmach** [comment]
the osm website does it as a string https://www.openstreetmap.org/api/0.6/node/1.json

so I guess it needs to be `interface{}`

**Comment by fawick** [comment]
Right, good idea!

And https://github.com/paulmach/osm/blob/master/osm.go#L302 could become

```go
o.Version = fmt.Sprint(s.Version)
```

Alternatively a type-switch to get around any IEEE754 weirdness
```go
switch x:=s.Version.(type) {
case float64:
    o.Version = fmt.Sprintf("%.1f", x.Version) 
case string:   
    o.Version = x
```

**Comment by captchanjack** [comment]
Seeing the same issue, I used this overpass query string

```
"[out:json];(way[\"highway\"][\"area\"!~\"yes\"][\"highway\"!~\"cycleway|footway|path|pedestrian|steps|track|corridor|elevator|escalator|proposed|construction|bridleway|abandoned|platform|raceway|service\"][\"motor_vehicle\"!~\"no\"][\"motorcar\"!~\"no\"][\"access\"!~\"private\"][\"service\"!~\"parking|parking_aisle|driveway|private|emergency_access\"](around:500,-37.740347,144.930127);>;relation[\"restriction\"][!\"conditional\"][!\"hgv\"][\"type\"=\"restriction\"](around:500,-37.740347,144.930127);>;)->._;out body;out count;out meta;"
```

**Comment by paulmach** [comment]
PR to fix this https://github.com/paulmach/osm/pull/46

**Comment by paulmach** [comment]
Merged and published [v0.7.1](https://github.com/paulmach/osm/releases/tag/v0.7.1)

Thank you for reporting this issue.
